### PR TITLE
feat(core): AWS config file

### PR DESCRIPTION
**Pull request by dnys1**
Adds support for the AWS config and credentials files, as defined in the SEP. This is critical for desktop support where configuration profiles are the main way credentials are managed.

### CODE REVIEW

**Review by haverchuck — COMMENTED** [review]
Can you please give a bit more detail on why this is critical for desktop support?

**Comment by dnys1** [comment]
Have you ever tried to use the Amplify or AWS CLI without a config file? If you do, you're gonna have a bad time lol. You're either copy/pasting creds into `export` statements or, in the case of Amplify, SOL.

It was the first thing Vlad asked about when I showed him the demo. For Windows, esp., environment variables are not as ubiquitous as on UNIX.

**Comment by haverchuck** [comment]
I understand why that is necessary for the CLIs, which require the developer's credentials to provision resources or call other operations which require developer credentials. I'm just trying to get clarity on why access to these credentials is necessary for our library.

**Comment by dnys1** [comment]
Well, so the signer is the only escape hatch we offer developers right now. In lieu of a full AWS SDK, it's the least we can do to provide proper ergonomics around that experience.

**Comment by haverchuck** [comment]
Ah ok, that makes sense.

**Review by Jordan-Nelson — APPROVED** [review]
lgtm. Did not test at all but test cases look pretty exhaustive.

**Comment by pytomtoto** [comment]
> 您能否详细说明为什么这对桌面支持至关重要？

I have multiple aws accounts ， I need switch accounts one by one。

File: packages/aws_common/lib/src/config/aws_config_value.dart [hunk 1]
```diff
@@ -16,31 +16,58 @@ import 'dart:async';
...
+  region._(
+    'AWS_REGION',
+    'us-east-1',
+  ),
+
+  /// The location (filepath) of the AWS configuration file.
+  ///
+  /// Defaults to `~/.aws/config`.
+  configFile._(
+    'AWS_CONFIG_FILE',
```
Comment (haverchuck) [inline packages/aws_common/lib/src/config/aws_config_value.dart:51]: Sorry, I think we talked about this at one point, but can you remind me what this is for?

Comment (dnys1) [reply packages/aws_common/lib/src/config/aws_config_value.dart:51]: No problem. It's a standard environment variable for setting the location of the config file: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list-AWS_CONFIG_FILE

**Review by haverchuck — APPROVED** [review]

**Pull request merged**